cmd/hypershift/info.go

```diff
@@ -0,0 +1,691 @@
+	infoCmd.Flags().StringVarP(&ops.output, "output", "o", "graphviz", "output format ['table', 'graphviz']")
+
+	return infoCmd
+}
+
+// infoOptions defines the struct for running the hypershift info command
+type infoOptions struct {
+	clusterID            string
+	awsProfile           string
+	awsRegion            string
+	privatelinkAccountId string
+	output               string
+	genericclioptions.IOStreams
+}
+
+// InfoClusters aggregates the OCM responses for the customer and management
+// cluster.
+type infoClusters struct {
+	managementCluster *v1.Cluster
+	customerCluster   *v1.Cluster
+}
+
+// Store all clients required to access the 3 different AWS accounts
+type hypershiftAWSClients struct {
+	customerClient    aws.Client
+	managementClient  aws.Client
+	privatelinkClient aws.Client
+}
+
+// Information related to the customer cluster
+type clusterInfo struct {
+	HostedZones       []*route53.HostedZone
+	ResourceRecords   []*route53.ResourceRecordSet
+	Endpoints         []*ec2.VpcEndpoint
+	Subnets           []*ec2.Subnet
+	SubnetRouteTables []*ec2.RouteTable
+}
+
+// Information related to the management cluster
+type managementClusterInfo struct {
+	EndpointConnections []*ec2.VpcEndpointConnection
+	EndpointServices    []*ec2.ServiceDetail
+	LoadBalancers       []*elbv2.LoadBalancer
+}
+
+// Information related to the privatelink account
+type privatelinkInfo struct {
+	HostedZones     []*route53.HostedZone
+	ResourceRecords []*route53.ResourceRecordSet
+	Endpoints       []*ec2.VpcEndpoint
+}
+
+// Aggregates all 3 AWS account informations in one
+type aggregateClusterInfo struct {
+	lock                  sync.Mutex
+	managementClusterInfo *managementClusterInfo
+	clusterInfo           *clusterInfo
+	privatelinkInfo       *privatelinkInfo
+}
+
+func newInfoOptions(streams genericclioptions.IOStreams) *infoOptions {
+	return &infoOptions{
+		IOStreams: streams,
+	}
+}
+
+func (i *infoOptions) complete(cmd *cobra.Command) error {
+	var errMsg string
+	if i.awsProfile == "" {
+		errMsg = "missing argument -p. "
+	}
+	if i.privatelinkAccountId == "" {
+		errMsg += "missing argument -l."
+	}
+	if i.output != "" {
+		if i.output != "graphviz" && i.output != "table" {
+			errMsg += "output must be 'graphviz' or 'table'"
+		}
+	}
+	if errMsg != "" {
+		return fmt.Errorf(errMsg)
+	}
+	return nil
+}
+
+func baseApiUrl(c *v1.Cluster) string {
+	return strings.Replace(strings.Replace(strings.Replace(c.API().URL(), "https://api.", "", 1), c.Name(), "", 1), ":443", "", 1)
+}
+
+func (i *infoOptions) run() error {
+	log.Println("Getting hypershift info for cluster: ", i.clusterID)
+	clusters, err := i.getClusters()
+	if err != nil {
+		return err
+	}
+	if i.awsRegion == "" {
+		i.awsRegion = clusters.customerCluster.Region().ID()
+		log.Println("Set aws region to: ", i.awsRegion)
+	}
+
+	log.Println("Constructing AWS sessions for all accounts")
```

Comment:
Is there a reason for using the `log` package here instead of `fmt`?  Most of the rest of osdctl isn't using log output, and while I'm not opposed, I'm not sure what the benefit of timestamps on a manually run too is.  It's valuable for actual log files if you're looking back at the history, but I think it just adds noise to the output for a CLI tool.

EDIT: Nevermind, looks like we're using log in a lot of places!  I take back my comment.

Reply:
Yea I guess I just looked at one of the commands that does, but I agree with the 'timestamps are not that useful' sentiment in this case - so I'm fine switching to just fmt printing, so we don't add more log code if we don't really want it.